test/sanity/pylint/ignore.txt

```diff
@@ -1,8 +1,16 @@
 lib/ansible/module_utils/network/iosxr/iosxr.py ansible-format-automatic-specification
 lib/ansible/modules/cloud/amazon/aws_api_gateway.py ansible-format-automatic-specification
+lib/ansible/modules/cloud/amazon/aws_config_aggregator.py duplicate-except
+lib/ansible/modules/cloud/amazon/aws_config_delivery_channel.py duplicate-except
+lib/ansible/modules/cloud/amazon/aws_config_recorder.py duplicate-except
+lib/ansible/modules/cloud/amazon/aws_config_rule.py duplicate-except
+lib/ansible/modules/cloud/amazon/aws_eks_cluster.py duplicate-except
 lib/ansible/modules/cloud/amazon/aws_kms.py ansible-format-automatic-specification
 lib/ansible/modules/cloud/amazon/ec2_eip.py ansible-format-automatic-specification
+lib/ansible/modules/cloud/amazon/ec2_vpc_vgw.py duplicate-except
 lib/ansible/modules/cloud/amazon/ecs_ecr.py ansible-format-automatic-specification
+lib/ansible/modules/cloud/amazon/rds_instance_facts.py duplicate-except
+lib/ansible/modules/cloud/amazon/rds_snapshot_facts.py duplicate-except
```

Comment:
Why are these ignores needed?

Reply:
Because the exception being caught is being dynamically generated and may or may not be the same class caught below. Pylint doesn't differentiate between this and static duplicates yet. I've opened a ticket in hopes that we can ignore only dynamically generate exception possible-duplicates in the future https://github.com/PyCQA/pylint/issues/2174 (so we don't accidentally mask static duplicates for which there is no reason).

Reply:
Why not add the error code check to the existing ClientError exception handlers instead of using dynamic types?

It looks like ClientError is already handled in most cases, so you'd end up with something like:

```python
    except botocore.exceptions.ClientError as e:
        if is_boto3_error_code('NoSuchConfigRuleException'):
            # do stuff here
        else:
            # do other stuff here
```

Reply:
Yep, it just simplifies things and gives us the equivalent to what we thought from_code was (interestingly, from_code is also catching duplicate exceptions, pylint isn't catching it somehow).

Allows:
```
except is_boto3_error_code('ExceptionCodeWithSpecialHandling'):
    pass
except (BotoCoreError, ClientError) as e:
    module.fail_json_aws(e, msg="error message")
```

rather than
```
except ClientError as e:
    if is_boto3_error_code('NoSuchConfigRuleException'):
        pass
    else:
        module.fail_json_aws(e, msg="error message")
except BotoCoreError as e:
    module.fail_json_aws(e, msg="error message")
```

Reply:
We should probably use inline pylint ignore comments for these instead of ignore.txt, since these are not issues that need fixing.

Reply:
ok

Reply:
Is the correct ignore the `# noqa` comment on the line? Or is there an ignore where you can specify the error?

Reply:
it's `# pylint: disable=duplicate-except`